staging/src/k8s.io/kube-aggregator/pkg/apiserver/handler_proxy.go

```diff
@@ -248,37 +244,38 @@ func (r *proxyHandler) updateAPIService(apiService *apiregistrationv1api.APIServ
 		return
 	}
 
-	newInfo := proxyHandlingInfo{
-		name: apiService.Name,
-		restConfig: &restclient.Config{
-			TLSClientConfig: restclient.TLSClientConfig{
-				Insecure:   apiService.Spec.InsecureSkipTLSVerify,
-				ServerName: apiService.Spec.Service.Name + "." + apiService.Spec.Service.Namespace + ".svc",
-				CertData:   r.proxyClientCert,
-				KeyData:    r.proxyClientKey,
-				CAData:     apiService.Spec.CABundle,
+	transportConfig := &transport.Config{
+		TLS: transport.TLSConfig{
+			Insecure: apiService.Spec.InsecureSkipTLSVerify,
+			GetCert: func() (*tls.Certificate, error) {
```

Comment:
could we unit test it? I think that `TestProxyHandler` tests start `TLS` server

Reply:
there are definitely tests already in place that were failing with tls errors during development so I kind of assume that's already covered. I still kind of feel like it would be nice to test the dynamic reload functionality 🤔 wondering what is the best level to test that though... any recommendations?